go/msgpack/msgpack.go

```diff
@@ -10,6 +10,13 @@ import (
 	"github.com/keybase/go-codec/codec"
 )
 
+func EncodeCanonical(src interface{}) (dst []byte, err error) {
+	ch := kbcrypto.CodecHandle()
+	ch.Canonical = true
```

Comment:
Is this something we have on everywhere? How does it differ from what we are currently doing?

Reply:
We do it in some places, like for box public summaries. The difference is just that every encoding of the same go struct will end with the same bytes. Without it I was running into trouble because when I reconstructed leafs, encoded and hashed them, they weren't matching the server's commitment.

Reply:
I don't understand this: "every encoding of the same go struct will end with the same bytes". What does this mean?